pageserver/src/config.rs

```diff
@@ -106,6 +110,168 @@ pub struct PageServerConf {
     pub remote_storage_config: Option<RemoteStorageConfig>,
 }
 
+// needed to simplify config construction
+struct PageServerConfigBuilder {
+    listen_pg_addr: Option<String>,
+
+    listen_http_addr: Option<String>,
+
+    checkpoint_distance: Option<u64>,
+    checkpoint_period: Option<Duration>,
+
+    gc_horizon: Option<u64>,
+    gc_period: Option<Duration>,
+    superuser: Option<String>,
+
+    page_cache_size: Option<usize>,
+    max_file_descriptors: Option<usize>,
+
+    workdir: Option<PathBuf>,
+
+    pg_distrib_dir: Option<PathBuf>,
+
+    auth_type: Option<AuthType>,
+
+    auth_validation_public_key_path: Option<Option<PathBuf>>,
+    remote_storage_config: Option<Option<RemoteStorageConfig>>,
```

Comment:
> `Option<Option<`

Out of curiosity, what does this semantic mean? 
Would we be able to achieve it with a single `Option<>` wrapper?

Reply:
Outer option is needed for builder to distinguish between set/not set, the inner one is the desired value in target structure we are building